secant/Features/WalletEventsFlow/Views/TransactionDetailView.swift

```diff
@@ -2,72 +2,160 @@ import SwiftUI
+                if let errorMessage = transaction.errorMessage {
+                    plainTwoColumnText(left: "Failed", right: errorMessage, mark: .fail)
+                }
             }
-            .padding()
 
-            Text("\(amountPrefixText) \(transaction.zecAmount.decimalString()) ZEC")
-                .transactionDetailRow()
+            Spacer()
             
-            Text("fee \(transaction.fee.decimalString()) ZEC")
-                .transactionDetailRow()
-
-            Text("total amount \(transaction.totalAmount.decimalString()) ZEC")
-                .transactionDetailRow()
+            footer
+        }
+        .applyScreenBackground()
+        .navigationTitle("Transaction detail")
+    }
+}
 
-            Button {
-                viewStore.send(.copyToPastboard(transaction.address))
-            } label: {
-                Text("\(addressPrefixText) \(transaction.address)")
-                    .lineLimit(1)
-                    .truncationMode(.middle)
-                    .transactionDetailRow()
+extension TransactionDetailView {
+    var header: some View {
+        HStack {
+            switch transaction.status {
+            case .pending:
+                Text("PENDING")
+                Spacer()
+            case .failed:
+                Text("\(transaction.date.asHumanReadable())")
+                Spacer()
+                Text("FAILED")
+            default:
+                Text("\(transaction.date.asHumanReadable())")
+                Spacer()
+                Text("HEIGHT \(heightText)")
             }
+        }
+        .padding()
+    }
+    
+    func plainText(_ text: String, mark: RowMark = .neutral) -> some View {
+        Text(text)
+            .transactionDetailRow(mark: mark)
+    }
 
-            if let memo = transaction.memo {
-                Button {
-                    viewStore.send(.copyToPastboard(memo))
-                } label: {
-                    VStack {
-                        Text("\(memo)")
-                            .multilineTextAlignment(.leading)
-                        
-                        HStack {
-                            Text("reply-to address included")
-                            Spacer()
-                            Button {
-                                viewStore.send(.replyTo(transaction.address))
-                            } label: {
-                                Text("reply now")
-                                    .padding(5)
-                                    .overlay(
-                                        RoundedRectangle(cornerRadius: 6)
-                                            .stroke(Asset.Colors.Text.transactionDetailText.color, lineWidth: 1)
-                                    )
-                            }
-                        }
+    func plainTwoColumnText(left: String, right: String, mark: RowMark = .neutral) -> some View {
```

Comment:
Shouldn't these be `@ViewBuilder` functions?

Reply:
No need to wrap it with @ViewBuilder. All functions are returning exact View type so SwiftUI doesn't need any helper (ViewBuilder, wrap to Group, wrap to AnyView or new struct View). We essentially have other views full or composable comonents 
```
// either
var header: some View

// or
func header() -> some View
```

As long as var or func is returning 1 specific type, SwiftUI is ok. When if/switch statements, one of the 4 mentioned solutions would be needed.